map_view/set-up-location-driven-geotriggers/src/main/java/com/esri/samples/set_up_location_driven_geotriggers/SetUpLocationDrivenGeotriggersController.java

```diff
@@ -0,0 +1,298 @@
+          setUpAndStartLocationDisplay();
+
+          // create feature fence parameters for the garden section, and for the garden points of interest with a buffer of 10m
+          FeatureFenceParameters featureFenceParametersGardenSection = new FeatureFenceParameters(gardenSectionFeatureTable);
+          FeatureFenceParameters featureFenceParametersPOI = new FeatureFenceParameters(gardenPOIFeatureTable, 10);
+
+          // define an arcade expression that returns the value for the "name" field of the feature that triggered the monitor
+          ArcadeExpression arcadeExpression = new ArcadeExpression("$fenceFeature.name");
+          // create a location geotrigger feed from the simulated location data source
+          LocationGeotriggerFeed locationGeotriggerFeed = new LocationGeotriggerFeed(simulatedLocationDataSource);
+
+          // create fence geotriggers for each of the service feature tables
+          FenceGeotrigger fenceGeotriggerGardenSection = new FenceGeotrigger(locationGeotriggerFeed, FenceRuleType.ENTER_OR_EXIT,
+            featureFenceParametersGardenSection, arcadeExpression, "Section Geotrigger");
+          FenceGeotrigger fenceGeotriggerPOI = new FenceGeotrigger(locationGeotriggerFeed, FenceRuleType.ENTER_OR_EXIT,
+            featureFenceParametersPOI, arcadeExpression, "POI Geotrigger");
+
+          // create geotrigger monitors from the fence geotriggers and start them
+          gardenSectionGeotriggerMonitor = new GeotriggerMonitor(fenceGeotriggerGardenSection);
+          gardenPOIGeotriggerMonitor = new GeotriggerMonitor(fenceGeotriggerPOI);
+
+          List<GeotriggerMonitor> geotriggerMonitors = new ArrayList<>(Arrays.asList(
+            gardenSectionGeotriggerMonitor, gardenPOIGeotriggerMonitor));
+          // for each geotrigger monitor, start it and add a notification listener
+          geotriggerMonitors.forEach(monitor -> {
+
+            monitor.startAsync();
+
+            monitor.addGeotriggerMonitorNotificationEventListener(notificationEvent -> {
+              // fence geotrigger notification info provides access to the feature that triggered the notification
+              var fenceGeotriggerNotificationInfo = (FenceGeotriggerNotificationInfo) notificationEvent.getGeotriggerNotificationInfo();
+
+              // get the name of the fence feature
+              fenceFeatureName = fenceGeotriggerNotificationInfo.getMessage();
+              geoTriggerName = fenceGeotriggerNotificationInfo.getGeotriggerMonitor().getGeotrigger().getName();
+              // determine the notification type on the notification info (entered or exited)
+              FenceNotificationType fenceNotificationType = fenceGeotriggerNotificationInfo.getFenceNotificationType();
+              ArcGISFeature fenceFeature = (ArcGISFeature) fenceGeotriggerNotificationInfo.getFenceGeoElement();
+
+              // when entering a given geofence, add the feature's information to the UI and save the feature for querying
```

Comment:
save the feature is bit confusing here, was that a reference to the earlier comment?